src/components/CIPipelineN/CIPipeline.tsx

```diff
@@ -191,18 +199,31 @@ export default function CIPipeline({
     })
     const validationRules = new ValidationRules()
     const [mandatoryPluginData, setMandatoryPluginData] = useState<MandatoryPluginDataType>(null)
+    const [pluginDataStore, setPluginDataStore] = useState<PluginDataStoreType>(
+        structuredClone(DEFAULT_PLUGIN_DATA_STORE),
+    )
+    const [availableTags, setAvailableTags] = useState<string[]>([])
+
     const selectedBranchRef = useRef(null)
 
     const mandatoryPluginsMap: Record<number, MandatoryPluginDetailType> = useMemo(() => {
         const _mandatoryPluginsMap: Record<number, MandatoryPluginDetailType> = {}
         if (mandatoryPluginData?.pluginData.length) {
             for (const plugin of mandatoryPluginData.pluginData) {
-                _mandatoryPluginsMap[plugin.id] = plugin
+                _mandatoryPluginsMap[plugin.parentPluginId] = plugin
             }
         }
         return _mandatoryPluginsMap
     }, [mandatoryPluginData])
 
+    const handlePluginDataStoreUpdate: PipelineContext['handlePluginDataStoreUpdate'] = (updatedPluginDataStore) => {
+        setPluginDataStore((prevPluginDataStore) => ({ ...prevPluginDataStore, ...updatedPluginDataStore }))
+    }
```

Comment:
We need to fix the discussed issue here as well. That's why i am suggesting moving the states to provider

Reply:
This here because code for CIPipeline and CDPipeline is following almost same structure and this is known issue but would be fixed through provider